File: andre/examples/c/module5/equal.c
#include <stdio.h>


typedef struct {
    int x, y;
} A;

typedef struct {
    int x, y;
} B;

void test(A a) {

    printf("a.x = %d\n", a.x);
    return;
}

int main() {

    B myB = {x: 1, y: 2};

    // A and B have the same internal structure. Can I use one for the other?
    test(myB);
    return 0;
}

File: andre/examples/go/module5/equal.go
package main

import "fmt"

type A struct {
	x int
	y int
}

type B struct {
	x int
	y int
}

func test(a A) {

	fmt.Printf("a.x = %d\n", a.x)
}

// refer to equal.c
func main() {

	var myB B = B{}

	test(myB);
}